expression/builtin_time_vec.go

```diff
@@ -313,11 +313,49 @@ func (b *builtinSubDateStringDecimalSig) vecEvalTime(input *chunk.Chunk, result
 }
 
 func (b *builtinPeriodDiffSig) vectorized() bool {
-	return false
+	return true
 }
 
+// evalInt evals PERIOD_DIFF(P1,P2).
+// See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_period-diff
 func (b *builtinPeriodDiffSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+	period1s := buf.Int64s()
+	period2s := buf1.Int64s()
+	result.ResizeInt64(n, false)
```

Comment:
We can reuse `result` buffer with `buf0` or `buf1`
Here's an [example](https://github.com/pingcap/tidb/blob/8de9c26805c5eb4be80fb3268180418a6c20e3b1/expression/builtin_math_vec.go#L631).